src/main/java/com/google/api/codegen/util/nodejs/NodeJSTypeTable.java

```diff
@@ -16,31 +16,76 @@
 
 import com.google.api.codegen.util.DynamicLangTypeTable;
 import com.google.api.codegen.util.NamePath;
+import com.google.api.codegen.util.TypeAlias;
+import com.google.api.codegen.util.TypeName;
 import com.google.api.codegen.util.TypeTable;
 import com.google.common.collect.ImmutableSet;
+import java.util.Map;
 
 /** The TypeTable for NodeJS. */
-public class NodeJSTypeTable extends DynamicLangTypeTable {
+public class NodeJSTypeTable implements TypeTable {
```

Comment:
@shinfan , it's better to look at the diff on the overall PR instead of this interdiff.